opencog/scm/CMakeLists.txt

```diff
@@ -33,6 +33,18 @@ ADD_GUILE_MODULE (FILES
 	MODULE_DESTINATION "${GUILE_SITE_DIR}/opencog"
 )
 
+# Create three copies of (opencog as-config) module:
+SET(SCM_IN_BUILD_DIR "#t")
+# One for build dir
+CONFIGURE_FILE(opencog/as-config_TEMPLATE.scm ${GUILE_BIN_DIR}/opencog/as-config.scm)
+# and one for in-place testing of source scm modules
+CONFIGURE_FILE(opencog/as-config_TEMPLATE.scm ${CMAKE_SOURCE_DIR}/opencog/scm/opencog/as-config.scm)
+
+SET(SCM_IN_BUILD_DIR "#f")
+# One for installing
+CONFIGURE_FILE(opencog/as-config_TEMPLATE.scm ${GUILE_BIN_DIR}/opencog/as-config-installable.scm)
+INSTALL(FILES ${GUILE_BIN_DIR}/opencog/as-config-installable.scm DESTINATION ${GUILE_SITE_DIR}/opencog RENAME as-config.scm)
```

Comment:
Why are you renaming during install ?  Why not just set the name in the first go?

Reply:
The installable config and build-dir config are different in terms of locating the shared libraries. Only `as-config-installable.scm` is actually installed. They both need to be configured and if they used the same name, one would clobber the other.

Reply:
Hmm OK.